[Tests] ChooseWorkspaceDialogTests.testFilterDuplicatedPaths fails on Windows in the I-Build

The test org.eclipse.ui.internal.ide.ChooseWorkspaceDialogTests.testFilterDuplicatedPaths failed on [windows](https://download.eclipse.org/eclipse/downloads/drops4/I20220928-1800/testresults/html/org.eclipse.ui.tests_ep426I-unit-win32-java11_win32.win32.x86_64_11.html) with the below error

```
Non-unique paths were not filtered as expected expected:<[some/location1/, /some/location1/, /some/location2/]> but was:<[some/location1/, /some/location1/, /some/location1, /some/location2/, /some///location1, //some//location1//]>

java.lang.AssertionError: Non-unique paths were not filtered as expected expected:<[some/location1/, /some/location1/, /some/location2/]> but was:<[some/location1/, /some/location1/, /some/location1, /some/location2/, /some///location1, //some//location1//]>
at org.junit.Assert.fail(Assert.java:89)
at org.junit.Assert.failNotEquals(Assert.java:835)
at org.junit.Assert.assertEquals(Assert.java:120)
at org.eclipse.ui.internal.ide.ChooseWorkspaceDialogTests.testFilterDuplicatedPaths(ChooseWorkspaceDialogTests.java:47) 
```

--- a/tests/org.eclipse.ui.tests/Eclipse UI Tests/org/eclipse/ui/internal/ide/ChooseWorkspaceDialogTests.java
+++ b/tests/org.eclipse.ui.tests/Eclipse UI Tests/org/eclipse/ui/internal/ide/ChooseWorkspaceDialogTests.java
@@ -15,6 +15,7 @@ package org.eclipse.ui.internal.ide;
 
 import static org.junit.Assert.assertEquals;
 
+import java.io.File;
 import java.util.Arrays;
 import java.util.List;
 
@@ -43,8 +44,18 @@ public class ChooseWorkspaceDialogTests {
 				"//some//location1//",
 		};
 		String[] expectedFilteredPaths = { "some/location1/", "/some/location1/", "/some/location2/", };
+		adoptToWindows(testPaths);
+		adoptToWindows(expectedFilteredPaths);
 		List<String> actualFilteredPaths = ChooseWorkspaceDialog.filterDuplicatedPaths(testPaths);
 		assertEquals("Non-unique paths were not filtered as expected", Arrays.asList(expectedFilteredPaths),
 				actualFilteredPaths);
+	}
+
+	static void adoptToWindows(String[] paths) {
+		if (File.separatorChar == '\\') {
+			for (int i = 0; i < paths.length; i++) {
+				paths[i] = paths[i].replace('/', '\\');
+			}
+		}
 	}
 }

